src/component/plotly-dash-preview/render.js

```diff
@@ -31,35 +33,53 @@ function render (info, opts, sendToMain) {
    * We check for a 'waitfor' div in the dash-app
    * which indicates that the app has finished rendering.
    */
-  const finishedLoading = () => {
+  const loaded = () => {
     return win.webContents.executeJavaScript(`
       new Promise((resolve, reject) => {
-        var a = document.getElementById("waitfor");
-        if (a) {
-          resolve(true);
+        let tries = ${cst.maxRenderingTries}
+
+        if (${info.timeOut}) {
+          tries = parseInt(${info.timeOut} * 1000/${cst.minInterval})
         }
+
+        let interval = setInterval(() => {
+          let el = document.querySelector("${info.loadingSelector}")
+
+          if (el) {
+            clearInterval(interval)
+            resolve(true)
+          }
+
+          if (--tries === 0) {
+            clearInterval(interval)
+
+            if (${info.timeOut}) {
+              resolve(true)
+            } else {
+              reject('fail to load')  
+            }
+          }
+        }, ${cst.minInterval})
+
       })`)
   }
 
   win.on('closed', () => {
     win = null
   })
 
-  let intervalId = setInterval(() => {
-    finishedLoading().then(loaded => {
-      if (loaded) {
-        clearInterval(intervalId)
-        contents.printToPDF(info.pdfOptions, (err, pdfData) => {
-          if (err) {
-            done(525)
-          } else {
-            result.imgData = pdfData
-            done()
-          }
-        })
+  loaded().then(() => {
+    contents.printToPDF(info.pdfOptions, (err, pdfData) => {
+      if (err) {
+        done(525)
+      } else {
+        result.imgData = pdfData
+        done()
       }
     })
-  }, 500)
+  }).catch(() => {
+    done(525)
```

Comment:
Sorry to not having noticed this before. But, it would be nice to separate `printToPDF` errors (that's the `done(525)` inside the `printToPDF` callback) and this one which is the result of a timeout. 

I propose making `525` the `printToPDF` callback error, and adding `526` when we hit a timeout.

Reply:
``` 
526: 'plotly.js version 1.11.0 or up required',
```
526 is already taken.. maybe traditional `408 Request Timeout` or `504 Gateway Timeout`?

Reply:
It's already taken in the `plotly-graph` component, right? Not in `plotly-dash-preview`? I don't like overriding 504 (a gateway timeout is something else) and I want this to be a 500-series.